### PR TITLE
api - redesign structure

### PR DESCRIPTION
**Pull request by Dkhanas**

### CODE REVIEW

File: tests/cli/test_dlab.py [hunk 1]
```diff
@@ -76,6 +78,8 @@ def setUp(self):
 
 class TestDLab(BaseCLITest):
 
+    # TODO check if sys.platform can help here for win
+    @unittest.skipIf(sys.platform == 'win32', reason="does not run on windows")
```
Comment (mediapills) [inline tests/cli/test_dlab.py:82]: Issue needs to be created and fixed in another PR

File: tests/unit/test_api.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# *****************************************************************************
```
Comment (mediapills) [inline tests/unit/test_api.py:1]: All of this file needs to be covered by API tests
tests needs to be in tests/api directory

File: tox.ini [hunk 3]
```diff
@@ -19,7 +19,7 @@ commands_pre =
     pip install -e plugins/dlab_aws
```
Comment (mediapills) [inline tox.ini:19]: I do not think you needs to use plugins here at all

File: dlab_core/app.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from flask import Flask
```
Comment (mediapills) [inline dlab_core/app.py:1]: better to use api/app.py location for this file because api is not a part of dlab_core library

File: dlab_core/schema_validator.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from jsonschema import validate
```
Comment (mediapills) [inline dlab_core/schema_validator.py:1]: this is abstract class will be better to put in  dlab_core/infrastructure

File: dlab_core/wsgi.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
```
Comment (mediapills) [inline dlab_core/wsgi.py:1]: will be better to move in /api/wsgi.py

File: plugins/dlab_templates/dlab_templates/infrastructure/controllers.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from dlab_core.infrastructure.controllers import BaseAPIController
+
+
+class APITemplatesController(BaseAPIController):
+    allowed_types = ['exploratory', 'computational']
```
Comment (mediapills) [inline plugins/dlab_templates/dlab_templates/infrastructure/controllers.py:5]: this needs to be a constants

File: plugins/dlab_templates/dlab_templates/infrastructure/controllers.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+    @classmethod
+    def get_templates(cls, type):
+        if type not in cls.allowed_types:
+            return {"code": 0, "message": "string"}, 400
+        return {}, 200
```
Comment (mediapills) [inline plugins/dlab_templates/dlab_templates/infrastructure/controllers.py:11]: codes needs to be a constants or you can get them from flask lib

File: plugins/dlab_templates/dlab_templates/infrastructure/views.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+from flask_restful import Resource
```
Comment (mediapills) [inline plugins/dlab_templates/dlab_templates/infrastructure/views.py:1]: this is not a view it is part of a routing as we discuss

**Review by mediapills — CHANGES_REQUESTED** [review]

**Comment by mediapills** [comment]
Could you please change branch name regards https://github.com/mediapills/dlab/blob/master/CONTRIBUTING.md ?

**Comment by Dkhanas** [comment]
move into new  [PR](https://github.com/mediapills/dlab/pull/131)

File: plugins/dlab_templates/dlab_templates/infrastructure/views.py (hunk 9 above)
Comment (Dkhanas) [reply plugins/dlab_templates/dlab_templates/infrastructure/views.py:1]: discussed

File: plugins/dlab_templates/dlab_templates/infrastructure/controllers.py (hunk 8 above)
Comment (Dkhanas) [reply plugins/dlab_templates/dlab_templates/infrastructure/controllers.py:11]: added statuses

File: plugins/dlab_templates/dlab_templates/infrastructure/controllers.py (hunk 7 above)
Comment (Dkhanas) [reply plugins/dlab_templates/dlab_templates/infrastructure/controllers.py:5]: Now constants

File: dlab_core/wsgi.py (hunk 6 above)
Comment (Dkhanas) [reply dlab_core/wsgi.py:1]: moved

File: dlab_core/schema_validator.py (hunk 5 above)
Comment (Dkhanas) [reply dlab_core/schema_validator.py:1]: moved

File: dlab_core/app.py (hunk 4 above)
Comment (Dkhanas) [reply dlab_core/app.py:1]: moved

File: tox.ini (hunk 3 above)
Comment (Dkhanas) [reply tox.ini:19]: removed

File: tests/unit/test_api.py (hunk 2 above)
Comment (Dkhanas) [reply tests/unit/test_api.py:1]: moved

File: tests/cli/test_dlab.py (hunk 1 above)
Comment (Dkhanas) [reply tests/cli/test_dlab.py:82]: removed
